### PR TITLE
Documents explicit type aliases

### PR DESCRIPTION
**Pull request by sobolevn**
Refs https://www.python.org/dev/peps/pep-0613/

I hope that I am correct with mypy's version number 🙂

### CODE REVIEW

File: docs/source/kinds_of_types.rst [hunk 1]
```diff
@@ -522,6 +521,21 @@ assigning the type to a variable:
...
+Because the distinction between an unannotated value and a type alias is implicit,
+ambiguous or incorrect type alias declarations implicitly default to a valid value assignment.
+
+.. code-block:: python
+
+   from typing import TypeAlias  # or `from typing_extensions` before `python3.10`
+
+   AliasType: TypeAlias = Union[list[dict[tuple[int, str], set[int]]], tuple[str, list[str]]]
+
+Explicit type aliases solves the ambiguness and improves readability.
```
Comment (JelleZijlstra) [inline docs/source/kinds_of_types.rst:537]:
```suggestion
Explicit type aliases are unambiguous and improve readability.
```
"ambiguess" is not a word :)

Comment (sobolevn) [reply docs/source/kinds_of_types.rst]: Thanks! Sometimes I make quite funny typos 🙂

**Review by JelleZijlstra — APPROVED** [review]

**Review by hauntsaninja — COMMENTED** [review]
Thanks, might also be worth leaving a note in https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases

**Comment by sobolevn** [comment]
@hauntsaninja done!

**Pull request merged**